private_set_intersection/python/__init__.py

```diff
@@ -1,21 +1,299 @@
+
+        return self.data.encrypted_elements()
+
+    def add_encrypted_elements(self, el: bytes):
+        """Add an item to the encrypted items in the request.
+        Args:
+            el: bytes buffer of the new item.
+        """
+        return self.data.add_encrypted_elements(el)
+
+    def clear_encrypted_elements(self):
+        """Clear the encrypted items in the request."""
+        return self.data.clear_encrypted_elements()
+
+    def reveal_intersection(self) -> bool:
+        """Get reveal_intersection flag value"""
+        return self.data.reveal_intersection()
+
+    def clear_reveal_intersection(self) -> bool:
+        """Reset reveal_intersection flag value"""
+        return self.data.clear_reveal_intersection()
+
+    def set_reveal_intersection(self, state: bool) -> bool:
+        """Set reveal_intersection flag value"""
+        return self.data.set_reveal_intersection(state)
+
+    def save(self) -> bytes:
+        """Save the protobuffer to wire format"""
+        return self.data.save()
+
+    def load(self, data: bytes):
+        """Load the protobuffer from wire format"""
+        return self.data.load(data)
+
+    @classmethod
+    def Load(cls, data: bytes) -> "proto_request":
+        """Load the protobuffer from wire format"""
+        return psi.cpp_proto_request.Load(data)
+
+
+class proto_response:
+    def __init__(self):
+        """Constructor method for the response Protobuf object.
+        Returns:
+            proto_response object.
+        """
+        self.data = psi.cpp_proto_response()
+
+    def encrypted_elements_size(self) -> int:
+        """Count of encrypted items in the response."""
+        return self.data.encrypted_elements_size()
+
+    def encrypted_elements(self, idx: int = None):
+        """Encrypted items in the response.
+        Args:
+            idx: If provided, returns the item at the idx position. Otherwise, returns all elements.
+        """
+        if idx is not None:
+            return self.data.encrypted_elements(idx)
+
+        return self.data.encrypted_elements()
+
+    def add_encrypted_elements(self, el: bytes):
+        """Add an item to the encrypted items in the response.
+        Args:
+            el: bytes buffer of the new item.
+        """
+        return self.data.add_encrypted_elements(el)
+
+    def clear_encrypted_elements(self):
+        """Clear the encrypted items in the request."""
+        return self.data.clear_encrypted_elements()
+
+    def save(self) -> bytes:
+        """Save the protobuffer to wire format"""
+        return self.data.save()
+
+    def load(self, data: bytes):
+        """Load the protobuffer from wire format"""
+        return self.data.load(data)
+
+    @classmethod
+    def Load(cls, data: bytes) -> "proto_request":
+        """Load the protobuffer from wire format"""
+        return psi.cpp_proto_response.Load(data)
+
+
+class client:
+    def __init__(self, reveal_intersection: bool, key_bytes: bytes = None):
```

Comment:
Are these default constructors really necessary for both client and server?

If this is very pythonic then we can leave it in, but it looks like we're just wrapping our class constructor functions with some defaults when instead we should have defaults provided into the class constructors themselves.

Ex: 
`psi.server.CreateWithNewKey(cls, reveal_intersection: bool = False)...`
`psi.server.CreateFromKey(cls, key_bytes: bytes, reveal_intersection: bool = False)...`

Reply:
Thank you for pointing this out.

Actually, the Python constructors didn't have any other role than populating the `data` member, and CreateWithNewKey/CreateFromKey were not doing that. 

I updated the PR.